tools/AzureRM.Example.psm1

```diff
@@ -9,6 +9,51 @@
 $PSDefaultParameterValues.Clear()
 Set-StrictMode -Version Latest
 
+function Test-DotNet
+{
+    try
+    {
+        if (Get-ChildItem 'HKLM:\SOFTWARE\Microsoft\NET Framework Setup\NDP\v4\Full\' -ErrorAction Stop | Get-ItemPropertyValue -Name Release | Foreach-Object { $_ -lt 461808 })
+        {
+            throw ".NET Framework versions lower than 4.7.2 are not supported in Az.  Please upgrade to .NET Framework 4.7.2 or higher."
```

Comment:
So, I don't think you want to catch this error, merely any errors that might occur from retrieving the registry key